llvm/lib/DebugInfo/DWARF/DWARFVerifier.cpp

```diff
@@ -167,20 +167,24 @@ bool DWARFVerifier::verifyUnitHeader(const DWARFDataExtractor DebugInfoData,
   if (!ValidLength || !ValidVersion || !ValidAddrSize || !ValidAbbrevOffset ||
       !ValidType) {
     Success = false;
-    error() << format("Units[%d] - start offset: 0x%08" PRIx64 " \n", UnitIndex,
-                      OffsetStart);
-    if (!ValidLength)
-      note() << "The length for this unit is too "
-                "large for the .debug_info provided.\n";
-    if (!ValidVersion)
-      note() << "The 16 bit unit header version is not valid.\n";
-    if (!ValidType)
-      note() << "The unit type encoding is not valid.\n";
-    if (!ValidAbbrevOffset)
-      note() << "The offset into the .debug_abbrev section is "
-                "not valid.\n";
-    if (!ValidAddrSize)
-      note() << "The address size is unsupported.\n";
+    ErrorCategory.Report("Invalid Length in Unit Header", [&]() {
```

Comment:
Awkward introducing new category names when there's nothing in the diagnostic itself that tells you it's part of a category (unlike, say, clang's diagnostic groups which are rendered in the diagnostic by default with the `[-Wunused]` or whatever after the diagnostic message) - could that be addressed in some way to make it clearer how these categories relate to the specific messages? (like if you wanted to grep for all the messages under a category to diff them against a baseline if you found the number went up or down)

Reply:
This was the only instance where something specifically like this occurred. My interest in not changing output with the aggregation trumped the "make this output searchable" goal. In an earlier revision, I'd reported each 'note' as a separate issue. I'd rather flag this with a TODO, so as not to change output yet, and have a subsequent diff that makes the output more consistent, and (as you mentioned) easier to connect to the category.

Reply:
I think, for now, I'll swap this back to individual categories to make it a bit easier to connect the category to the unique error.